src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -824,6 +825,14 @@ int goto_instrument_parse_optionst::doit()
       else
         return CPROVER_EXIT_SUCCESS;
     }
+    else if(cmdline.args.size() < 2)
+    {
+      throw invalid_command_line_argument_exceptiont(
+        "Invalid number of positional arguments passed",
+        "[in] [out]",
+        "goto-instrument needs one input and one output file, aside from other "
+        "flags");
```

Comment:
You could also maybe add an additional line here to say "use goto-instrument --help for additional help" or something similar.